### ISSUE TITLE
Add all necessary contracts to the SystemConfig’s `Addresses` struct (OPCM - Upgrade 13)

### ISSUE THREAD

**Issue by maurelian** [ethereum-optimism/optimism#13073]
The OPCM uniquely identifies each OP Chain by its `SystemConfig`. For upgrades the OPCM will retrieve the `Addresses` struct from the `SystemConfig` to identify all contracts it needs to interact with during the upgrade process.

Currently that struct is lacking certain contracts which are required, including: 
- AddressManager
- DelayedWeth(s) 
	- although these might be best retrieved by querying: `gameImpl(DisputeGameFactory.gameImpls(n)).weth()`

A new getter should also be added to the `SystemConfig` to retrieve the full `Addresses` struct in a single call.

**Comment by mds1** [comment]
I think the AddressManager is actually not accessible onchain, though I recall @smartcontracts might have had an idea for how to access it? https://github.com/ethereum-optimism/client-pod/issues/699

For DelayedWETH, I agree with using the `gameImpl(DisputeGameFactory.gameImpls(n)).weth()` approach. We might want to have a new struct that returns addresses that are not part of the existing `Addresses` struct?

**Comment by maurelian** [comment]
> I think the AddressManager is actually not accessible onchain

This is true, we'll need to provide that address as an input to this upgrade.


> For DelayedWETH, I agree with using the gameImpl(DisputeGameFactory.gameImpls(n)).weth() approach. We might want to have a new 
struct that returns addresses that are not part of the existing Addresses struct?

To help me understand: what do you think that struct looks like, what else is in it? what would it be called?

**Comment by maurelian** [comment]
> I think the AddressManager is actually not accessible onchain

Turns out it's very easy to get from `ProxyAdmin.addressManager()`.

**Comment by maurelian** [comment]
I reviewed the full set of contracts deployed by `OPCM.deploy()`, here is what I documented: 

| OPCM Deploys                         | System Config Has?             | Notes / Discoverable from                                    | Action? |
| ------------------------------------ | ------------------------------ | ------------------------------------------------------------ | ------- |
| `opChainProxyAdmin`                  | No                             | `UpgradeInput` (as currently planned), but could be avoided. | None    |
| `addressManager`                     | No                             | Discover from: `ProxyAdmin`                                  | None    |
| `l1ERC721BridgeProxy`                | `l1ERC721Bridge`               |                                                              | None    |
| `systemConfigProxy`                  | No                             | `UpgradeInput`                                               | None    |
| `optimismMintableERC20FactoryProxy`  | `optimismMintableERC20Factory` |                                                              | None    |
| `l1StandardBridgeProxy`              | `l1StandardBridge`             |                                                              | None    |
| `l1CrossDomainMessengerProxy`        | `l1CrossDomainMessenger`       |                                                              | None    |
| `optimismPortalProxy`                | `optimismPortal`               |                                                              | None    |
| `disputeGameFactoryProxy`            | `disputeGameFactory`           |                                                              | None    |
| `faultDisputeGame`                   | No                             | Discoverable from: `disputeGameFactory.gameImpls(0)`.        | None    |
| `permissionedDisputeGame`            | No                             | Discoverable from: `disputeGameFactory.gameImpls(1)`.        | None    |
| `anchorStateRegistryProxy`           | No                             | Discoverable from: `faultDisputeGame.anchorStateRegistry()`  | None    |
| `anchorStateRegistryImpl`            | No                             | Will be deployed by `OPCM.upgrade()`                         | None    |
| `delayedWETHPermissionlessGameProxy` | No                             | Discoverable from: `permissionlessDisputeGame.weth()`        | None    |
| `delayedWETHPermissionedGameProxy`   | No                             | Discoverable from: `permissionedDisputeGame.weth()`          | None    |


In this table `UpgradeInput` is a placeholder for anything which is currently planned to be an input to `OPCM.upgrade(ISystemConfig[], IProxyAdmin[])`. 

So, we could simplify future upgrades by adding `ProxyAdmin` address to the `SystemConfig.Addresses` struct, so that we only need to provide a single argument to the upgrade function per chain, however this is a nice to have, so I think that we can remove this from scope.
